RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js

```diff
@@ -64,25 +36,33 @@ function init() {
         let ws = new WebSocket(wsProtocol + location.host + path + "websocket");
 
         ws.binaryType = 'arraybuffer';
-    
+
         ws.onopen = function() {
             console.log("RGBME WebSocket connection established!");
             startTime = performance.now();
             requestImage();
         };
 
         ws.onclose = function() {
-            // Handle retries via requestImage call
-            requestImage();
+            // Handle retries by recreating the connection to websocket.
+            console.warn(`RGBME WebSocket connection lost. Retrying in ${WS_RETRY_DELAY / 1000}s.`)
+            setTimeout(function() {
+                // We generate socket with a timeout to make sure server has time to recover.
+                socket = generateSocket();
+            }, 2000);
+        }
+
+        ws.onerror = function() {
+            ws.close();
```

Comment:
The `close` method is important to not cause memory leaks.

Reply:
And apparently this might be not enough according to this SO answer. I'll check. In my tests I had no issues with it. 

https://stackoverflow.com/a/11982071

Reply:
I think if we run into these edge cases in the future we can optimize it then

Most people use the emulator as a dev tool so it's on their local machine and short-lived.